src/Map/StructuredPopup.php

```diff
@@ -29,7 +29,7 @@ private function getPropertyValueList(): string {
 		$lines = [];
 
 		foreach ( $this->propertyValues as $name => $value ) {
-			$lines[] = $this->bold( $this->stripTags( $name ) ) . ': ' . $this->stripTags( $value );
+			$lines[] = $this->bold( $this->stripTags( $name ) ) . ': ' . $this->stripTags( strval( $value ) );
```

Comment:
The constructor takes `@param string[] $propertyValues`, so this should not be needed. Either some caller has an issue, or perhaps all callers can provide null, in which case the issue is with the constructor.

Reply:
Good point, you're right this should be fine.

And I'm sorry I rather rushed this PR I think! I jumped at a simple solution, but I should've reported the actual error — which seems to come from interaction between Cargo and Maps (with PHP 8.1, and current master of core):

```
[da366c5a198e10b441fa067c] Cargo TypeError: Maps\Map\StructuredPopup::stripTags():
Argument #1 ($html) must be of type string, null given, called in
mediawiki/extensions/Maps/src/Map/StructuredPopup.php on line 32

Backtrace:

from /home/sam/code/wikimedia/mediawiki/extensions/Maps/src/Map/StructuredPopup.php(38)
#0 mediawiki/extensions/Maps/src/Map/StructuredPopup.php(32): Maps\Map\StructuredPopup->stripTags()
#1 mediawiki/extensions/Maps/src/Map/StructuredPopup.php(22): Maps\Map\StructuredPopup->getPropertyValueList()
#2 mediawiki/extensions/Maps/src/Map/CargoFormat/CargoOutputBuilder.php(124): Maps\Map\StructuredPopup->getHtml()
#3 mediawiki/extensions/Maps/src/Map/CargoFormat/CargoOutputBuilder.php(91): Maps\Map\CargoFormat\CargoOutputBuilder->newMarker()
#4 mediawiki/extensions/Maps/src/Map/CargoFormat/CargoOutputBuilder.php(50): Maps\Map\CargoFormat\CargoOutputBuilder->cargoValuesToMarkers()
#5 mediawiki/extensions/Maps/src/Map/CargoFormat/CargoOutputBuilder.php(40): Maps\Map\CargoFormat\CargoOutputBuilder->createMarkers()
#6 mediawiki/extensions/Maps/src/Map/CargoFormat/CargoFormat.php(30): Maps\Map\CargoFormat\CargoOutputBuilder->buildOutputFromCargoData()
#7 mediawiki/extensions/Cargo/includes/CargoQueryDisplayer.php(391): Maps\Map\CargoFormat\CargoFormat->display()
#8 mediawiki/extensions/Cargo/includes/parserfunctions/CargoQuery.php(168): CargoQueryDisplayer->displayQueryResults()
#9 mediawiki/includes/parser/Parser.php(3453): CargoQuery::run()
#10 mediawiki/includes/parser/Parser.php(3136): Parser->callParserFunction()
#11 mediawiki/includes/parser/PPFrame_Hash.php(276): Parser->braceSubstitution()
#12 mediawiki/includes/parser/Parser.php(2971): PPFrame_Hash->expand()
#13 mediawiki/includes/parser/Parser.php(1607): Parser->replaceVariables()
#14 mediawiki/includes/parser/Parser.php(723): Parser->internalParse()
#15 mediawiki/includes/content/WikitextContentHandler.php(295): Parser->parse()
#16 mediawiki/includes/content/ContentHandler.php(1714): WikitextContentHandler->fillParserOutput()
#17 mediawiki/includes/content/Renderer/ContentRenderer.php(47): ContentHandler->getParserOutput()
#18 mediawiki/includes/Revision/RenderedRevision.php(266): MediaWiki\Content\Renderer\ContentRenderer->getParserOutput()
#19 mediawiki/includes/Revision/RenderedRevision.php(237): MediaWiki\Revision\RenderedRevision->getSlotParserOutputUncached()
#20 mediawiki/includes/Revision/RevisionRenderer.php(221): MediaWiki\Revision\RenderedRevision->getSlotParserOutput()
#21 mediawiki/includes/Revision/RevisionRenderer.php(158): MediaWiki\Revision\RevisionRenderer->combineSlotOutput()
#22 [internal function]: MediaWiki\Revision\RevisionRenderer->MediaWiki\Revision\{closure}()
#23 mediawiki/includes/Revision/RenderedRevision.php(199): call_user_func()
#24 mediawiki/includes/poolcounter/PoolWorkArticleView.php(91): MediaWiki\Revision\RenderedRevision->getRevisionParserOutput()
#25 mediawiki/includes/poolcounter/PoolWorkArticleViewCurrent.php(97): PoolWorkArticleView->renderRevision()
#26 mediawiki/includes/poolcounter/PoolCounterWork.php(162): PoolWorkArticleViewCurrent->doWork()
#27 mediawiki/includes/page/ParserOutputAccess.php(299): PoolCounterWork->execute()
#28 mediawiki/includes/page/Article.php(702): MediaWiki\Page\ParserOutputAccess->getParserOutput()
#29 mediawiki/includes/page/Article.php(517): Article->generateContentOutput()
#30 mediawiki/includes/actions/ViewAction.php(78): Article->view()
#31 mediawiki/includes/MediaWiki.php(550): ViewAction->show()
#32 mediawiki/includes/MediaWiki.php(322): MediaWiki->performAction()
#33 mediawiki/includes/MediaWiki.php(916): MediaWiki->performRequest()
#34 mediawiki/includes/MediaWiki.php(570): MediaWiki->main()
#35 mediawiki/index.php(50): MediaWiki->run()
#36 mediawiki/index.php(46): wfIndexMain()
#37 {main}
```

Reply:
Then it looks like `buildOutputFromCargoData` in `CargoOutputBuilder` is getting null values in the array. Bug in Cargo? Or expected?

Either way, seems like something to deal with in `CargoOutputBuilder`

Reply:
@samwilson have you decided to not follow up on this?

Reply:
No, it's still on my todo list. But it looks like it can be closed here, and taken up on the Cargo side. Sorry for leaving this hanging!

Reply:
I've reported this as https://phabricator.wikimedia.org/T323861